iree/hal/cuda/CMakeLists.txt

```diff
@@ -82,6 +83,22 @@ iree_cc_library(
   PUBLIC
 )
 
+if(IREE_HAL_DRIVER_CUDA AND IREE_ENABLE_RUNTIME_TRACING)
+iree_cc_test(
+  NAME
+    dynamic_cupti_test
+  SRCS
+    "dynamic_cupti_test.cc"
+  DEPS
+    ::dynamic_symbols
+    iree::base
+    iree::testing::gtest
+    iree::testing::gtest_main
+  LABELS
+    "driver=cuda"
+)
+endif() # //IREE_HAL_DRIVER_CUDA AND IREE_ENABLE_RUNTIME_TRACING
```

Comment:
CMake actually supports an argument to endif that exactly matches the if condition. It's kind of a legacy thing that used to be required and isn't recommended anymore, but if you think it's helpful then I would do it that way. The main reason those comments are useful in C++ preprocessor are because it doesn't have proper indenting though and you could indent here (although I guess not if you add this on the Bazel side as well)